### DISCUSSION TITLE
Add multiple inventories for movable items

### DISCUSSION THREAD

**Discussion by DeinococcusR** [FakeFishGames/Barotrauma#4827] [Ideas and suggestions]
Would allow a lot of modding ideas to have multiple inventories in movable items. Like e.g. you could have diving suits with cargo space and weapons that need different sorts of ammo to work to name only two.

**Comment by Sviat** [comment]
Too complex

**Comment by mLuby** [comment]
I agree, this would unlock tons of modding. More precisely, it's allowing multiple independent `<ItemContainer/>` elements on a single item such that child elements (like `<StatusEffect/>`) respect the separation. For example:

```xml
<Item id="divingsuit">
  <ItemContainer capacity="1">
    <!-- When an O2 tank is contained, use it to keep the character oxygenated. -->
    <Containable items="oxygentank" />
    <StatusEffect … />
  </ItemContainer>
  <ItemContainer capacity="1">
    <!-- When a battery is contained, use it to power the suit's headlamp. -->
    <Containable items="mobilebattery" />
    <StatusEffect … />
  </ItemContainer>
</Item>
```

**Reply by Sviat** [reply]
and how you visually separate them? We already have Welding fuel tanks in diving suits every once in a while.
If slot is going to be constrained, then it needs proper error message to user, but makes things complex.

**Reply by mLuby** [reply]
Players learn what goes where by trial and error as it is currently, but if you wanted to I think item inventory GUIs can have labels, like "Large Steel Cabinet" or my preferred solution: using the item icon ghosts (C) from the fabricator to indicate what belongs where. ![](https://camo.githubusercontent.com/1f0ccd7b76324cd1d34f790339aca7037604b656fdf6d7973b18c8fa2981da2a/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3232303838313731383636363732333332382f3830303733363030323238343036303638322f4c6162656c6c65645f436f6e7461696e65722e706e67)

**Reply by Sviat** [reply]
Ghosts would be okay, if it would be another Fabricator with exact items.
But, for example, Oxygen could be supplied by two different cans

**Comment by NilanthAnimosus** [comment]
I recall when having items like diving suits use multiple slots, there was a problem with being able to have the same item (IE. an oxygen tank) which would drain both tanks dry as status effects on contained will apply to each item individually at the same time (Think of an oxygen generator recharging oxygen tanks, the same applies to a 2-slot diving suit with 2 oxygen tanks). which may seem counter intuitive if you started adding more slots, making this more needlessly complicated.

I am also not sure how well the UI handles having multiple, separate containers on the same item while wearing it accessing those slots, in order to prevent the above and differentiate the slots, adding yet again - additional complexity.

It would make more sense to simply give worn clothing like engineer's hazmat suits / engineer jumpsuits their own item slots, than to add them to something like a diving suit.

As for multi-items-required to actually shoot weapons. I'd rather just give my weapon a durability bar and a way to fix it as even that would be less needlessly complex than filling my gun with parts and bullets to fire it (Or multiple bullet pieces but, again - I'd just make the bullet one singular item at that point). IE. if your concept on weapons was to add gunpowder and lead balls to a blunderbuss like gun, I'd rather craft the gunpowder and lead balls together and just drop those into the gun directly.

Additions need to be simple and intuitive, Maybe the slots could be given small image variations depending on what it can contain, and proper separation to prevent dual batteries/oxygen tanks etc implemented somehow, but then you are just adding more work and items required to dive as an additional chore (more battery recharging and carrying batteries) - I'm picturing my split now between oxygen tanks and batteries alone to loot a wreck and the slots I will have remaining to do so.

**Reply by mLuby** [reply]
> would drain both tanks dry as status effects on contained will apply to each item individually at the same time (Think of an oxygen generator recharging oxygen tanks, the same applies to a 2-slot diving suit with 2 oxygen tanks).

Yes, this is a good example of what I meant by saying the child elements must respect the separation. I consider that they _don't_ respect the separation to be a bug. 🐞

I checked—you're right that the GUI does not currently support multiple separate item inventories for a single item. That feels rather unexpected/unintended, if you ask me. 🐞
